src/main/java/org/jabref/gui/maintable/OpenUrlAction.java

```diff
@@ -62,11 +67,28 @@ public void execute() {
 
             if (link.isPresent()) {
                 try {
-                    JabRefDesktop.openExternalViewer(databaseContext, link.get(), field);
+                    if (field.equals(StandardField.DOI) && preferences.getDOIPreferences().isUseCustom()) {
+                        openCustomDOI(link.get());
+                    } else {
+                        JabRefDesktop.openExternalViewer(databaseContext, link.get(), field);
+                    }
                 } catch (IOException e) {
                     dialogService.showErrorDialogAndWait(Localization.lang("Unable to open link."), e);
                 }
             }
         });
     }
+
+    private void openCustomDOI(String link) {
```

Comment:
I saw that this method is a duplicate of the one in the Entry Editor. To avoid code duplication, I think it could be useful to extract this method to  a static method in JabRefDesktop and then call that.  You would need to pass the prefs as parameter then